### PR TITLE
Optionally  save logs in storage

### PR DESCRIPTION
**Pull request by miiu96**
Added an extra flag in **config.toml** file to can `enable` or `disable` if logs generated by the transactions are saved in storage.

_In the case of a node with DBLookUpExtensions logs will be saved by default in storage ( no matter if the flag is set to true of false)_

### CODE REVIEW

File: factory/processComponents.go [hunk 1]
```diff
@@ -263,9 +263,16 @@ func (pcf *processComponentsFactory) Create() (*processComponents, error) {
 	}
 
 	txLogsStorage := pcf.data.StorageService().GetStorer(dataRetriever.TxLogsUnit)
+
+	if !pcf.config.LogsAndEvents.SaveInStorageEnabled && pcf.config.DbLookupExtensions.Enabled {
```
Comment (iulianpascalau) [inline factory/processComponents.go:267]: :+1:

File: process/transactionLog/process_test.go [hunk 2]
```diff
@@ -64,6 +64,39 @@ func TestTxLogProcessor_SaveLogsEmptyLogsReturnsNil(t *testing.T) {
...
+		Storer:      &testscommon.StorerStub{},
+		Marshalizer: &mock.MarshalizerMock{},
+	})
+
+	logs := []*vmcommon.LogEntry{
+		{Address: []byte("first log")},
+	}
+	err := txLogsProc.SaveLog([]byte("txhash"), &transaction.Transaction{}, logs)
+	require.Nil(t, err)
+
```
Comment (iulianpascalau) [inline process/transactionLog/process_test.go:92]: maybe remove empty lines?

Comment (miiu96) [reply process/transactionLog/process_test.go]: done.

**Review by sasurobert — APPROVED** [review]

**Review by iulianpascalau — COMMENTED** [review]
:+1:

**Review by iulianpascalau — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**